SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -102,13 +126,15 @@ - (void)sdl_accessoryDisconnected:(NSNotification *)notification {
     EAAccessory *accessory = [notification.userInfo objectForKey:EAAccessoryKey];
     if (accessory.connectionID == self.session.accessory.connectionID) {
         self.sessionSetupInProgress = NO;
+        [self sdl_backgroundTaskEnd];
         [self disconnect];
         [self.delegate onTransportDisconnected];
     }
 }
 
 - (void)sdl_applicationWillEnterForeground:(NSNotification *)notification {
     [SDLDebugTool logInfo:@"App Foregrounded Event" withType:SDLDebugType_Transport_iAP toOutput:SDLDebugOutput_All toGroup:self.debugConsoleGroupName];
+    [self sdl_backgroundTaskEnd];
```

Comment:
Based on all of my research, this is something handled by Apple when we go into the foreground. I have not seen any sample code in which the task is ended on foregrounding. The only thing I have seen happen is when the app becomes active, that the background task is started if it was ever ended. The sample code I have looked at has the background task end in the expiration handler and whenever the task is fully completed. Let me know if you've seen anything describing usage differently.